test/C/src/Deadline.lf

```diff
@@ -14,11 +14,7 @@ reactor Source(period: time(3 sec)) {
         if (2 * (self->count / 2) != self->count) {
             // The count variable is odd.
             // Take time to cause a deadline violation.
-            // Do not use nanosleep() here because if this is threaded,
-            // it gets awakened right away.
```

Comment:
I checked the documentation of the sleep functions that our platform abstraction uses, and I did not find any support for this comment. I'm not sure why it was here for so long, but thanks for removing it.

Reply:
I'm fine with the removal of the comment, but I think that `nanosleep` returns prematurely if the thread receives a signal.

Reply:
Yes, that's true. Is that likely to happen in the typical running of our tests?

Reply:
I think it may have been the reason that the busy-wait approach was opted for.

Reply:
If the only signals we expect are those that signify critical errors or requests to terminate the program, then I would not worry about them. However, I do not have a strong opinion on this issue.

Reply:
Since the deadline tests pass it means that the sleep is long enough. I made some local tests and indeed `lf_nanosleep` does not return prematurely (in my tests). While this does not give a guarantee, it shows at least that the existing comment is wrong (it does not get awakened right away). We could put the sleep in a loop, but it would do much more sense to do this in the implementation of `lf_nanosleep` itself.